### ISSUE TITLE
Consider adding an API that takes the nonce size and tag size as parameters

### ISSUE THREAD

**Issue by d3lm** [RustCrypto/AEADs#544]
We are building a Node.js runtime for the browser so you can run Node.js entirely in your browser without relying on cloud servers. That means we run arbitrary code and right now it's only possible to specify the nonce size and tag size statically via types so we are forced to always call `from_slice(...)`. Ideally we can create it once and not rely on a match that matches nonce size and tag size, e.g.

```rust
 match nonce.len() {
   12 => match tag_size {
     12 => AesGcm::<Aes, U12, U12>::new_from_slice(key),
     ...
   },
   ...
 }
```

**Comment by tarcieri** [comment]
Alternatively an operation like `encrypt_in_place_detached` could accept an additional `&mut [u8]` slice to write the tag into, returning an error if its length isn't in the range `12..=16` (see #541 regarding lengths of 4 and 8, which require special care to support correctly)

**Comment by d3lm** [comment]
Hm but that means that the tag size still needs to be configured statically right?

**Comment by tarcieri** [comment]
No, it would take a slice, and return an error if it wasn't a valid size
